src/i18n/i18n.php

```diff
@@ -170,20 +171,9 @@ public static function _t($entity, $arg = null)
             user_error("Missing default for localisation key $entity", E_USER_WARNING);
         }
 
-        // Deprecate legacy injection format (`string %s, %d`)
-        // inject the variables from injectionArray (if present)
-        $sprintfArgs = [];
-        if ($default && !preg_match('/\{[\w\d]*\}/i', $default ?? '') && preg_match('/%[s,d]/', $default ?? '')) {
-            $sprintfArgs = array_values($injection ?? []);
-            $injection = [];
-        }
-
-        // If injection isn't associative, assume legacy injection format
-        $failUnlessSprintf = false;
-        if ($injection && array_values($injection ?? []) === $injection) {
-            $failUnlessSprintf = true; // Note: Will trigger either a deprecation error or exception below
-            $sprintfArgs = array_values($injection ?? []);
-            $injection = [];
+        // Injection must be associative
+        if (!empty($injection) && !ArrayLib::is_associative($injection)) {
+            throw new InvalidArgumentException('Injection must be an associative array');
         }
```

Comment:
Uses a more declarative condition than the previous one, but the result is the same.

Error moved here from below, as per the comment (see line 209)